File: C/guessing_game.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void)
{
    int n;
    int step = 0;
    int guess;
    char ch;

    srand(time(NULL));

    n = (rand() % 100) + 1;

    puts("You should guess the number in range 1 .. 100");

    do
    {
        printf("Enter your guess: ");
        if (scanf("%d", &guess) != 1)
        {
            puts("You entered bullshit :(");
            while (ch = getchar(), ch != '\n' && ch != EOF)
                ;
            continue;
        }

        if (guess < n)
        {
            puts("Your guess is too low :(");
        }
        else if (guess > n)
        {
            puts("Your guess is too high :(");
        }

        step++;
    } while (guess != n);

    printf("You won in %d steps!\n", step);

    return EXIT_SUCCESS;
}

File: Julia/guessing_game.jl
n = rand(1:100)
step = 1

println("You should guess the number in range 1 .. 100")

while true
    print("Enter your guess: ")
    guess = parse(Int, readline())

    if guess < n
        println("Your guess is too low :(")
    elseif guess > n
        println("Your guess is too high :(")
    else
        break
    end

    global step += 1
end

println("You won in ", step, " steps!")
